pkg/statsd/receiver.go

```diff
@@ -79,13 +96,19 @@ func (mr *MetricReceiver) Receive(ctx context.Context, c net.PacketConn) {
 			continue
 		}
 		// TODO consider updating counter for every N-th iteration to reduce contention
```

Comment:
If we are reading a lot of packets toady, this may actually make sense. In a separate PR.

Reply:
This was one of the first things I tested, and it didn't seem to make a difference.  With the changes here it's even less important because we basically get the "every N-th" for free.